src/v/archival/segment_reupload.cc

```diff
@@ -193,25 +193,26 @@ segment_collector::find_next_compacted_segment(model::offset start_offset) {
     if (it == segment_set.end()) {
         vlog(
           archival_log.debug,
-          "Finding next compacted segment for {}: can't find segment after "
+          "Finding next segment for {}: can't find segment after "
           "offset: {}",
           _manifest.get_ntp(),
           start_offset);
         return {};
     }
 
     const auto& segment = *it;
-    if (segment->finished_self_compaction()) {
+    auto is_compacted = segment->finished_self_compaction();
```

Comment:
Variable names are confusing here: "is_compacted" refers to whether the segment has been compacted yet, but non_compacted refers to the configuration of the topic.  Maybe change these to is_segment_compacted and is_topic_compacted or similar.